### PR TITLE
Internal: Create infrastructure for the implementation of the Home screen [ED-13724]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: includes/settings/settings.php [hunk 1]
```diff
@@ -83,10 +87,26 @@ public function register_admin_menu() {
...
+				$this,
+				$this->home_module->is_experiment_active() ? 'display_home_screen' : 'display_settings_page',
+			],
 			'',
 			'58.5'
 		);
+
+		if ( $this->home_module->is_experiment_active() ) {
+			add_action( 'elementor/admin/menu/register', function( Admin_Menu_Manager $admin_menu ) {
+				$admin_menu->register( 'settings', new Admin_Menu_Item( $this ) );
```
Comment (louiswol94) [inline includes/settings/settings.php:100]: Why `settings` as the Page ID?

File: includes/settings/settings.php [hunk 2]
```diff
@@ -438,6 +462,8 @@ public function add_generator_tag_settings( $settings ) {
 	public function __construct() {
 		parent::__construct();
 
+		$this->home_module = new Home_Module();
```
Comment (louiswol94) [inline includes/settings/settings.php:465]: Might be a good chance to try Dependency Injection ;)

File: modules/home/assets/js/app.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+					<HomeScreen />
+				</ThemeProvider>
+			</LocalizationProvider>
+		</DirectionProvider>
+	);
+};
+
+const isRTL = elementorCommon.config.isRTL,
+	rootElement = document.querySelector( '#e-home-screen' ),
+	colorScheme = rootElement?.dataset?.ui_theme || 'auto';
```
Comment (louiswol94) [inline modules/home/assets/js/app.js:21]:
```suggestion
	colorScheme = elementor?.getPreferences?.( 'ui_theme' ) || 'auto';
```
Maybe?

File: tests/bootstrap.php [hunk 4]
```diff
@@ -66,6 +66,7 @@
 add_action( 'elementor/experiments/feature-registered', function ( Experiments_Manager $experiments_manager, array $experimental_data ) {
 	$exclude = [
 		Editor::EDITOR_V2_EXPERIMENT_NAME, // For now the tests are not ready for the new editor.
+		'home_screen',
```
Comment (louiswol94) [inline tests/bootstrap.php:69]:
```suggestion
		Elementor\Modules\Home\Module::PAGE_ID,
```
Maybe?

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/home/assets/js/app.js (hunk 3 above)
Comment (hein-obox) [reply modules/home/assets/js/app.js:21]: @louiswol94 This isn't working. I think `elementor` isn't available outside of the editor. I wasn't able to get it to work.

File: includes/settings/settings.php (hunk 1 above)
Comment (hein-obox) [reply includes/settings/settings.php:100]: In the current setup, the settings page uses `elementor` as the Page ID, but when the experiment is active, the settings page should get its own Page Id, and the top level 'elementor' Page ID, will be used for the new home screen.

**Review by louiswol94 — APPROVED** [review]

**Review by murzin-nikita — APPROVED** [review]

**Pull request merged**